### PR TITLE
Fixes watch tracking during reloads and fixes address issue.

### LINKED ISSUE #3177: Unable to register watch handlers (regression with 0.8.4)
If you have a question, please direct it to the
[consul mailing list](https://www.consul.io/community.html) if it hasn't been
addressed in either the [FAQ](https://www.consul.io/docs/faq.html) or in one
of the [Consul Guides](https://www.consul.io/docs/guides/index.html).

When filing a bug, please include the following:

### `consul version` for both Client and Server
Client: 0.8.4
Server: master

### Description of the Issue (and unexpected/desired result)
Creating watches fail with the error ```[ERR] Failed to run watch: %v Failed to connect to agent: Unknown protocol scheme: http+tcp```
### Reproduction steps
Run consul with a watch handler configuration https://www.consul.io/docs/agent/watches.html. I will link to the one I used in a gist as a comment below.  
### Log Fragments or Link to [gist](https://gist.github.com/)
 ```[ERR] Failed to run watch: %v Failed to connect to agent: Unknown protocol scheme: http+tcp```
Include appropriate Client or Server log fragments.  If the log is longer
than a few dozen lines, please include the URL to the
[gist](https://gist.github.com/).

TIP: Use `-log-level=TRACE` on the client and server to capture the maximum log detail.

### PR DESCRIPTION
**Pull request by slackpad**
Fixes #3177.
Closes #3181.
Closes #3186.

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -2316,50 +2325,28 @@ func (a *Agent) ReloadConfig(newCfg *Config, prevConfig *Config) (bool, error) {
...
-		errs = multierror.Append(errs, fmt.Errorf("Failed reloading metadata: %s", err))
-		return false, errs
+		return fmt.Errorf("Failed reloading metadata: %s", err)
 	}
 
-	// Get the new client listener addr
-	httpAddr, err := newCfg.ClientListener(newCfg.Addresses.HTTP, newCfg.Ports.HTTP)
-	if err != nil {
-		errs = multierror.Append(errs, fmt.Errorf("Failed to determine HTTP address: %v", err))
+	// Reload the watches.
```
Comment (magiconair) [inline agent/agent.go]: Comment is redundant.

File: agent/agent_endpoint_test.go [hunk 2]
```diff
@@ -261,19 +264,16 @@ func TestAgent_Reload(t *testing.T) {
...
-		t.Fatalf("got ok %v want true")
-	}
 	if _, ok := a.state.services["redis-reloaded"]; !ok {
 		t.Fatalf("missing redis-reloaded service")
 	}
-	//verify that previous config's watch plans were stopped
-	for _, w := range cfg.WatchPlans {
-		if !w.IsStopped() {
+
+	// Verify that previous config's watch plans were stopped.
```
Comment (magiconair) [inline agent/agent_endpoint_test.go]: Comment is redundant.

File: agent/agent.go (hunk 1 above)
Comment (slackpad) [inline agent/agent.go]: Removed in https://github.com/hashicorp/consul/commit/4a3604a3eefbff4c7245d3790d26a37f5d5aa206.

File: agent/agent_endpoint_test.go (hunk 2 above)
Comment (slackpad) [inline agent/agent_endpoint_test.go]: Removed in https://github.com/hashicorp/consul/commit/4a3604a3eefbff4c7245d3790d26a37f5d5aa206.

**Pull request merged**